### PR TITLE
Migrate Improvements: nav bar

### PR DESCRIPTION
**Pull request by joekarasek**
Worked through this one with Evan, all the updates seem pretty straightforward to us, but wanted to put this up in it's own PR so we could see the changes up on Netlify. How's it look? Seems that these are @remydenton updates; mind taking a peak?

### CODE REVIEW

File: packages/components/bolt-nav-bar/src/nav-bar.standalone.js [hunk 1]
```diff
@@ -6,28 +6,108 @@ import {
...
     // Ensure that 'this' inside the _onWindowResize event handler refers to <bolt-nav-link>
     // even if the handler is attached to another element (window in this case)
     this._onWindowResize = this._onWindowResize.bind(this);
   }
 
-  renderCallback() {
-    return (
-      <slot />
+  render() {
+    if (hasNativeShadowDomSupport) {
```
Comment (sghoweri) [inline packages/components/bolt-nav-bar/src/nav-bar.standalone.js]: FYI @joekarasek @remydenton @EvanLovely @rockymountainhigh1943 @mikemai2awesome lines 78 through 81 and 85 through 91 are a great example of one of the handful of use cases I've been working to tighten up and standardize across the board (and am getting close to having a PRable solution).

Basically, any Web Component that needs to pass along nested children should do so via the natively supported <slot> tag (including named slots) for content projection and in browsers that don't support it or in the handful of situations that we want to opt components out of using Shadow DOM, we automatically render the child content where it needs to go without requiring an extra polyfill or requiring spaghetti code workarounds.

Basically, I want us to have a single way to handle this across the board like https://stenciljs.com/docs/templating does and eliminate the overhead from component authoring entirely
